Src/Lib/ResourceHelper.cpp

```diff
@@ -476,8 +476,19 @@ HBITMAP LoadImageFile( const wchar_t *path, const SIZE *pSize, bool bUseAlpha, b
 			frameWidthD=abs(pSize->cx);
 			if (pSize->cy)
 				frameHeightD=pSize->cy;
+			else if (maxSize && maxSize->cy > 0)
+			{
+				float aspectRatio = (float)frameWidthS / (float)frameHeightS;
+				frameHeightD = frameWidthD * frameHeightS / frameWidthS;
+				if (frameHeightD > maxSize->cy) {
+					frameHeightD = maxSize->cy;
+					frameWidthD = frameHeightD * aspectRatio;
```

Comment:
This could be just:
`frameWidthD = frameHeightD * frameWidthS / frameHeightS;`

It will avoid needless _float_ operation as we need to obtain integer value (also it will avoid the compilation warning).